core/vm/contracts.go

```diff
@@ -1096,3 +1115,35 @@ func (c *pointEvaluation) RequiredGas(input []byte) uint64 {
 func (c *pointEvaluation) Run(input []byte) ([]byte, error) {
 	return kzg.PointEvaluationPrecompile(input)
 }
+
+type parentBeaconBlockRoot struct{}
+
+func (c *parentBeaconBlockRoot) RequiredGas(input []byte) uint64 {
+	return params.ParentBeaconBlockRootGas
+}
+
+func (c *parentBeaconBlockRoot) Run(input []byte) ([]byte, error) {
+	return nil, nil
+}
+
+func (c *parentBeaconBlockRoot) RunStateful(input []byte, state evmtypes.IntraBlockState) ([]byte, error) {
+	timestampParam := input[:32]
+	if len(timestampParam) < 32 {
+		return nil, errors.New("timestamp param too short")
+	}
+
+	timestampReduced := uint256.NewInt(0).SetBytes(timestampParam).Uint64() % params.HistoricalRootsModulus
+	timestampIndex := libcommon.BigToHash(libcommon.Big256.SetUint64((timestampReduced)))
+	recordedTimestamp := uint256.NewInt(0)
+	root := uint256.NewInt(0)
+	state.GetState(libcommon.BytesToAddress(params.HistoryStorageAddress), &timestampIndex, recordedTimestamp)
+
+	recordedTimestampBytes := recordedTimestamp.Bytes32()
+	if !bytes.Equal(recordedTimestampBytes[:], timestampParam) {
+		return make([]byte, 32), nil
+	}
+	timestampExtended := timestampReduced + params.HistoricalRootsModulus
+	rootIndex := libcommon.BigToHash(libcommon.Big256.SetUint64((timestampExtended)))
+	state.GetState(libcommon.BytesToAddress(params.HistoryStorageAddress), &rootIndex, root)
+	return root.Bytes(), nil
```

Comment:
Should be `root.Bytes32()` instead since `root.Bytes()` truncates leading zeros.